From f7e83ba8cd92df261ab1ded77a11a678d842668f Mon Sep 17 00:00:00 2001
From: Wang Wensheng <wangwensheng4@huawei.com>
Date: Thu, 7 Jul 2022 11:35:44 +0000
Subject: [PATCH 3300/4585] [Huawei] mm/sharepool: Fix kabi borken in
 sp_group_master

Offering: HULK
ascend inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I5DS9S

-------------------------------------------------

The sp_group_master structure is used only in sharepool subsys and no
other drivers use it.

Signed-off-by: Wang Wensheng <wangwensheng4@huawei.com>
---
 include/linux/share_pool.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/include/linux/share_pool.h b/include/linux/share_pool.h
index 25b84d995619..1911cd35843b 100644
--- a/include/linux/share_pool.h
+++ b/include/linux/share_pool.h
@@ -10,6 +10,7 @@
 #include <linux/hashtable.h>
 #include <linux/numa.h>
 #include <linux/jump_label.h>
+#include <linux/kabi.h>
 
 #define SP_HUGEPAGE		(1 << 0)
 #define SP_HUGEPAGE_ONLY	(1 << 1)
@@ -187,7 +188,7 @@ struct sp_group_master {
 	 * Used to apply for the shared pool memory of the current process.
 	 * For example, sp_alloc non-share memory or k2task.
 	 */
-	struct sp_group *local;
+	KABI_EXTEND(struct sp_group *local)
 };
 
 /*
-- 
2.17.1

